src/schema/types.rs

```diff
@@ -74,6 +74,43 @@ impl Type {
     }
   }
 
+  /// Check if 'sub_type' schema is part of current schema, e.g. projected columns
+  pub fn check_contains(&self, sub_type: &Type) -> bool {
+    // names match, and repetitions match or not set for both
+    let basic_match = self.get_basic_info().name() == sub_type.get_basic_info().name() && (
+      self.is_schema() && sub_type.is_schema() || !self.is_schema() && !sub_type.is_schema() &&
+      self.get_basic_info().repetition() == sub_type.get_basic_info().repetition()
+    );
+
+    match *self {
+      Type::PrimitiveType { .. } if basic_match && sub_type.is_primitive() => {
+        self.get_physical_type() == sub_type.get_physical_type()
+      },
+      Type::GroupType { .. } if basic_match && sub_type.is_group() => {
+        // build hashmap of name -> TypePtr
+        let mut field_map = HashMap::new();
+        for field in self.get_fields() {
+          field_map.insert(field.name(), field);
+        }
+
+        for field in sub_type.get_fields() {
+          if !field_map.get(field.name()).map(|tpe| tpe.check_contains(field)).unwrap_or(false) {
+            return false;
+          }
+        }
+        true
+      },
+      _ => false,
+    }
+  }
+
+  fn get_physical_type(&self) -> PhysicalType {
```

Comment:
Is this okay to have this function for `Type` enum?

Reply:
Yes it is OK. Perhaps better to move it after `get_fields()`.